integration-tests/http/__tests__/product/store/product.spec.ts

```diff
@@ -401,5 +469,640 @@ medusaIntegrationTestRunner({
         expect(response.status).toEqual(200)
       })
     })
+
+    describe("GET /store/products", () => {
+      beforeEach(async () => {
+        inventoryItem1 = (
+          await api.post(
+            `/admin/inventory-items`,
+            { sku: "test-sku" },
+            adminHeaders
+          )
+        ).data.inventory_item
+
+        inventoryItem2 = (
+          await api.post(
+            `/admin/inventory-items`,
+            { sku: "test-sku-2" },
+            adminHeaders
+          )
+        ).data.inventory_item
+        ;[product, [variant]] = await createProducts({
+          title: "test product 1",
+          status: ProductStatus.PUBLISHED,
+          variants: [
+            {
+              title: "test variant 1",
+              manage_inventory: true,
+              inventory_items: [
+                {
+                  inventory_item_id: inventoryItem1.id,
+                  required_quantity: 20,
+                },
+                {
+                  inventory_item_id: inventoryItem2.id,
+                  required_quantity: 20,
+                },
+              ],
+              prices: [{ amount: 3000, currency_code: "usd" }],
+            },
+          ],
+        })
+        ;[product2, [variant2]] = await createProducts({
+          title: "test product 2 uniquely",
+          status: ProductStatus.PUBLISHED,
+          variants: [
+            { title: "test variant 2", manage_inventory: false, prices: [] },
+          ],
+        })
+        ;[product3, [variant3]] = await createProducts({
+          title: "product not in price list",
+          status: ProductStatus.PUBLISHED,
+          variants: [{ title: "test variant 3", prices: [] }],
+        })
+        ;[product4, [variant4]] = await createProducts({
+          title: "draft product",
+          status: ProductStatus.DRAFT,
+          variants: [{ title: "test variant 4", prices: [] }],
+        })
+
+        const defaultSalesChannel = await createSalesChannel(
+          { name: "default sales channel" },
+          [product.id, product2.id, product3.id, product4.id]
+        )
+
+        const service = appContainer.resolve(ModuleRegistrationName.STORE)
```

Comment:
Let's move this to a `beforeEach`, and let's not store the `container` globally as we are trying to avoid people calling modules directly if possible

Reply:
same